gc.c

```diff
@@ -162,6 +162,8 @@ static char *mmtk_env_plan = NULL;
 static char *mmtk_pre_arg_plan = NULL;
 static char *mmtk_post_arg_plan = NULL;
 static char *mmtk_chosen_plan = NULL;
+
+size_t *mmtk_max_heap_size = NULL;
```

Comment:
I guess this is a pointer because you want it to be optional. I think in C it's more idiomatic to have a separate flag to say if it's set and then a value. Otherwise we need to manage the lifetime of the value when it's set. (I know the lifetime here is simple because it's a big global, but in general.)

Should also be `static`, as it's not used outside this file.

Reply:
Thanks Chris. This pointer was actually just a copy paste error on my part. I've replaced it with just a `size_t` and then relying on whether it's > 0 to determine whether it's been set or not. On the assumption that `--mmtk-max-heap=0` is  not something users would set intentionally.